x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.js

```diff
@@ -754,4 +760,79 @@ export class DataRecognizer {
     return false;
   }
 
+  applyJobConfigOverrides(moduleConfig, jobOverrides, jobPrefix = '') {
+    if(jobOverrides !== undefined) {
+      if (typeof jobOverrides !== 'object') {
+        throw Boom.badRequest(
+          `Incompatible jobOverrides type (${typeof jobOverrides}). It needs to be an object or array of objects.`
+        );
+      }
+
+      // jobOverrides could be a single object or an array of objects.
+      // if single, convert to an array
+      const overrides = Array.isArray(jobOverrides) ? jobOverrides : [jobOverrides];
+      const { jobs } = moduleConfig;
+
+      // first collect the overrides which don't contain a job id
+      // these will be applied to all jobs in the module
+      const generalOverrides = overrides.filter(o => o.job_id === undefined);
```

Comment:
Couldn't this loop over `overrides` just once to separate out those with job ids and those without instead of running filter on `overrides` twice?